src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -203,25 +195,8 @@ export const UnderlineNav = forwardRef(
       return breakpoint
     }
 
-    const [selectedLink, setSelectedLink] = useState<RefObject<HTMLElement> | undefined>(undefined)
-
-    // selectedLinkText is needed to be able set the selected menu item as selectedLink.
-    // This is needed because setSelectedLink only accepts ref but at the time of setting selected menu item as selectedLink, its ref as a list item is not available
-    const [selectedLinkText, setSelectedLinkText] = useState<string>('')
-    // Capture the mouse/keyboard event when a menu item is selected so that we can use it to fire the onSelect callback after the menu item is swapped with the list item
-    const [selectEvent, setSelectEvent] = useState<
-      React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement> | null
-    >(null)
-
```

Comment:
This was a bad solution 🙈 I implemented this a way back and now looking at it now, triggering onSelect function on the selected menu item is as straightforward as [this](https://github.com/primer/react/blob/2a3887a84d9581933a122939931f1799920334af/src/UnderlineNav2/UnderlineNav.tsx#L376)